### PR TITLE
Add UnixFilePermissions.xml for Mono AOT compilers

### PR DESCRIPTION
**Pull request by directhex**
As is, mono-aot-cross files installed to a system location via workloads are not executable by the end user:

```
-rwxr--r--  1 root  wheel    11M Jun  1 14:33 /usr/local/share/dotnet/packs/Microsoft.NETCore.App.Runtime.AOT.osx-x64.Cross.browser-wasm/6.0.0-preview.5.21301.5/tools/mono-aot-cross
```

Ref: https://github.com/dotnet/runtime/issues/53545
Ref: https://github.com/dotnet/sdk/issues/16894
Ref: https://github.com/xamarin/xamarin-macios/pull/11869
Ref: https://github.com/xamarin/xamarin-android/pull/6010

### CODE REVIEW

File: src/installer/pkg/sfx/Microsoft.NETCore.App/Microsoft.NETCore.App.MonoCrossAOT.UnixFilePermissions.xml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<FileList>
+    <File Path="tools/mono-aot-cross" Permission="755" />
+    <File Path="tools/opt" Permission="755" />
+    <File Path="tools/llc" Permission="755" />
```
Comment (radical) [inline src/installer/pkg/sfx/Microsoft.NETCore.App/Microsoft.NETCore.App.MonoCrossAOT.UnixFilePermissions.xml:4]: Do these need .exe for windows?

**Review by radical — APPROVED** [review]
Other than the .exe comment, LGTM!

**Review by rolfbjarne — APPROVED** [review]

**Comment by rolfbjarne** [comment]
Does this have to be back ported to P6?

File: src/installer/pkg/sfx/Microsoft.NETCore.App/Microsoft.NETCore.App.MonoCrossAOT.UnixFilePermissions.xml (hunk 1 above)
Comment (jonathanpeppers) [reply src/installer/pkg/sfx/Microsoft.NETCore.App/Microsoft.NETCore.App.MonoCrossAOT.UnixFilePermissions.xml:4]: I don't think `UnixFilePermissions.xml` is needed at all on Windows.

File: src/installer/pkg/sfx/Microsoft.NETCore.App/Microsoft.NETCore.App.MonoCrossAOT.sfxproj [hunk 2]
```diff
@@ -20,6 +20,7 @@
   <ItemGroup>
     <NativeRuntimeAsset Include="$(MonoAotCrossDir)$(TargetCrossRid)\**" TargetPath="tools/" />
     <NativeRuntimeAsset Include="$(MonoAotCrossDir)$(TargetCrossRid).Sdk.props" TargetPath="Sdk/Sdk.props" />
+    <NativeRuntimeAsset Include="Microsoft.NETCore.App.MonoCrossAOT.UnixFilePermissions.xml" TargetPath="data/UnixFilePermissions.xml" />
```
Comment (jonathanpeppers) [inline src/installer/pkg/sfx/Microsoft.NETCore.App/Microsoft.NETCore.App.MonoCrossAOT.sfxproj:23]: Can you omit this file for Windows packages?

Comment (directhex) [reply src/installer/pkg/sfx/Microsoft.NETCore.App/Microsoft.NETCore.App.MonoCrossAOT.sfxproj:23]: That is a reasonable point

**Comment by directhex** [comment]
@rolfbjarne good question. I think the answer is "yes", the supporting changes in dotnet/sdk are in P6

**Pull request merged**